packages/core/test/actions.test.ts

```diff
@@ -725,9 +727,115 @@ describe('entry/exit actions', () => {
         'exit: a.a1',
         'exit: a',
         'enter: a',
+        'enter: a.a1',
+        'enter: a.a1.a11'
+      ]);
+    });
+
+    it('should reenter leaf state during its self-transition', () => {
```

Comment:
This test and the other ones added below were already working. I was trying to "break" the existing implementation for those cases but they seem to work fine. They might overlap with some existing tests but I've figured out that it's better to have more than fewer tests so I've committed them 😜